src/magicgui/backends/_qtpy/widgets.py

```diff
@@ -433,12 +442,48 @@ def _mgui_get_text(self) -> str:
         """Get text."""
         return self._qwidget.text()
 
+    def _update_icon(self) -> None:
+        # Called when palette changes or icon is set
+        if self._icon:
+            qicon = _get_qicon(*self._icon, palette=self._qwidget.palette())
+            if qicon is None:
+                self._icon = None  # an error occurred don't try again
+                self._qwidget.setIcon(QIcon())
+            else:
+                self._qwidget.setIcon(qicon)
+
+    def _mgui_set_icon(self, value: str | None, color: str | None) -> None:
+        self._icon = (value, color)
+        self._update_icon()
+
+
+def _get_qicon(key: str | None, color: str | None, palette: QPalette) -> QIcon | None:
+    """Return a QIcon from iconify, or None if it fails."""
+    if not key:
+        return QIcon()
+
+    if not color or color == "auto":
+        # use foreground color
+        color = palette.color(QPalette.ColorRole.WindowText).name()
+        # don't use full black or white
+        color = {"#000000": "#333333", "#ffffff": "#cccccc"}.get(color, color)
+
+    if ":" not in key:
+        # for parity with the other backends, assume fontawesome
+        # if no prefix is given.
+        key = f"fa:{key}"
+
+    try:
+        return superqt.QIconifyIcon(key, color=color)
+    except (OSError, ValueError) as e:
+        warnings.warn(f"Could not set iconify icon: {e}", stacklevel=2)
+        return None
```

Comment:
Does this will handle HTTP errors if someone try to use it the first time on a machine without internet access?

Reply:
Yeah I believe it should (that was my intention anyway).  Requests HTTP errors are subclasses of OSError